app/lib/services/simulationPasses/SimulationPassService.js

```diff
@@ -43,23 +86,28 @@ class SimulationPassService {
      * @returns {Promise<SimulationPass[]>} Promise object represents the result of this use case.
      */
     async getSimulationPassesPerPeriod(periodId, query) {
-        const baseClause = {
-            include: [
-                {
-                    model: Period,
-                    required: true,
-                    attributes: [],
-                    through: {
+        const baseClause = deepmerge(commonClause,
+            {
+                include: [
+                    {
+                        model: Period,
+                        required: true,
+                        attributes: [],
                         where: {
-                            period_id: periodId,
+                            id: periodId,
+                        },
+                        through: {
+                            attributes: [],
                         },
                     },
-                },
-            ],
-        };
+                ],
+            });
 
         const { count, rows } = await SimulationPassRepository.findAndCountAll(new QueryBuilder(baseClause).addFromHttpRequestQuery(query));
-        return { count, rows: rows.map((simulationPass) => simulationPassAdapter.toEntity(simulationPass)) };
+        return {
+            count: count.length,
```

Comment:
`count.length`?

Reply:
yes, because this query returns list of counts per one grouping, so in this query `count := Array<Object<{id: <Number>, count: <Number>}>>`, so it is list of counts of related data associated with one simulation pass (id of data pass) the same applies for other PR's with statements like that

Reply:
Indeed. Sequelize 🤮